packages/server/lib/browsers/firefox.ts

```diff
@@ -676,6 +681,21 @@ export async function open (browser: Browser, url: string, options: BrowserLaunc
       return browserReturnStatus || driverReturnStatus
     }
 
+    // In some cases, the webdriver session will NOT return the moz:debuggerAddress capability even though
+    // we set it to true in the capabilities. This is out of our control, so when this happens, we fail the browser
+    // and gracefully terminate the related processes and attempt to relaunch the browser in the hopes we get a
+    // CDP address.
+    if (!webdriverClient.capabilities['moz:debuggerAddress']) {
```

Comment:
@AtofStryker Did you bring this to the Firefox teams attention?

Reply:
I asked about it on the main issue just to see if this has been seen before https://github.com/cypress-io/cypress/issues/30352#issuecomment-2405701867

Reply:
Can we link to that comment inline so that we can potentially follow up?

Reply:
added in c7cc208b49